### DISCUSSION TITLE
alsa-lib stops on loading alsa-plugins installed with apt

### DISCUSSION THREAD

**Discussion by tomoyanonymous** [Homebrew/discussions#790] [Linux]
Hello.
My bottled tap package failed to launch when alsa-lib tries to load plugins with the messages like this.

```log
ALSA lib conf.c:3725:(snd_config_hooks_call) Cannot open shared library libasound_module_conf_pulse.so (/home/linuxbrew/.linuxbrew/Cellar/alsa-lib/1.2.4/lib/alsa-lib/libasound_module_conf_pulse.so: cannot open shared object file: No such file or directory)
ALSA lib control.c:1471:(snd_ctl_open_noupdate) Invalid CTL default
```

As a temporary solution, `sudo apt remove libasound2-plugins` solved the error, though this disables PulseAudio and other plugins completely on the system...

A relevant issue is reported in this thread. The difficult part seems that the plugin directory is a compile-time constant and cannot be changed with the environment variables...

https://github.com/NixOS/nixpkgs/issues/6860

Confirmed on Ubuntu 20.04 (also another user on Manjaro reported this for me).

**Comment by danielnachun** [comment]
Are you trying to run this in a `linuxbrew` install whose cellar is not `/home/linuxbrew/.linuxbrew`?  If `libasound` has compile time constants hardcoded as strings in the binary, then the bottle is not relocatable under our current system, and must be built from source if the install prefix is not `/home/linuxbrew/.linuxbrew`.  

I should note that I have a plan in the works to implement the binary patching scheme used by Anaconda to Homebrew, but I don't have any timeline of when this would happen.  AFAIK Anaconda is the only package manager right now which supports full binary relocation without the use of some type of `chroot` or VM.

**Reply by tomoyanonymous** [reply]
No, it is installed in `/home/linuxbrew/.linuxbrew/Cellar/alsa-lib/1.2.4` as the error message shows. `/home/linuxbrew/.linuxbrew/Cellar/alsa-lib/1.2.4/lib` does not contain `alsa-lib` directory.
However, I could not reproduce this issue after once I reinstalled with `build-from-source` option even I reinstalled it with `--force-bottle` option and reinstalled `libasound2-plugins `again...

**Comment by directentis1** [comment]
I got the same problem on my Xubuntu 20.04 when I tried to run a audio tool (`sox` player) on my terminal.
I also tried to rebuild `alsa-lib` with option `build-from-source` and `--force-bottle` but nothing changes.

How could you solved it then?

![](https://i.imgur.com/yJZOhcM.png)

**Reply by directentis1** [reply]
I think I've solved this by install pulseaudio package with homebrew, though I hadn't run homebrew-pulseaudio as service (cause it conflicts with apt-pulseaudio).
FYI.
